lib/fluent/plugin/in_windows_eventlog2.rb

```diff
@@ -258,19 +269,26 @@ def subscribe_channels(subscriptions)
       end
     end
 
-    def bookmark_validator(bookmarkXml, channel)
-      return false if bookmarkXml.empty?
+    def bookmark_valid?(bookmarkXml, channel)
+      if @@bookmark_parser_avaiable
+        bookmark_valid_strictly?(bookmarkXml, channel)
+      else
+        bookmarklist_is_not_empty?(bookmarkXml, channel)
+      end
+    end
 
+    def bookmark_valid_strictly?(bookmarkXml, channel)
       evtxml = WinevtBookmarkDocument.new
       parser = Nokogiri::XML::SAX::Parser.new(evtxml)
       parser.parse(bookmarkXml)
       result = evtxml.result
-      if !result.empty? && (result[:channel].downcase == channel.downcase) && result[:is_current]
-        true
-      else
-        log.warn "This stored bookmark is incomplete for using. Referring `read_existing_events` parameter to subscribe: #{bookmarkXml}, channel: #{channel}"
-        false
-      end
+      !result.empty? && (result[:channel].downcase == channel.downcase) && result[:is_current]
+    end
+
+    def bookmarklist_is_not_empty?(bookmarkXml, channel)
+      # Empty example: "<BookmarkList>\r\n</BookmarkList>"
+      # Not empty example: "<BookmarkList>\r\n  <Bookmark Channel='Setup' RecordId='777' IsCurrent='true'/>\r\n</BookmarkList>"
+      bookmarkXml.include?("Channel")
```

Comment:
Although it's better than depending nokogiri mandatory, it might be better to do more something here.

* Try rexml
* Match not only `Channel` but also `<BookmarkList>` and `</BookmarkList>` to check the file isn't truncated.

In addition, I've noticed that winevt_c should check `GetLastError()` when it fails on `EvtCreateBookmark()`: https://github.com/fluent-plugins-nursery/winevt_c/blob/9dd9c81432b1e0180c7abc19de8415bb42db2e49/ext/winevt/winevt_bookmark.c#L93

Reply:
e.g.) https://learn.microsoft.com/en-us/windows/win32/wes/bookmarking-events

Reply:
It's worth to merge even without ↑
We'll tackle on it in another PR.

Reply:
Thanks! Sorry I could not make time for this.
Yes, we'll tackle on it in another PR.